torch/_export/__init__.py

```diff
@@ -246,4 +246,4 @@ def do_not_use_experimental_export(f: Callable, args: Tuple, training=False):
 # )
 def dynamic_dim(t: torch.Tensor, index: int):
     from torch._dynamo.eval_frame import Constraint
-    return Constraint(weakref.ref(t), id(t), index, StrictMinMaxConstraint(ValueRanges(lower=2, upper=sympy.oo)))
+    return Constraint(weakref.ref(t), id(t), index, StrictMinMaxConstraint(vr=ValueRanges(lower=2, upper=sympy.oo), warn_only=False))
```

Comment:
Oof, the field reordering is a pain. New suggestion: don't do the superclass, instead just manually add `warn_only` to each dataclass (and if you want, define a Protocol that requires these classes to define `warn_only`.

Reply:
I think the superclass is a better idea tho. This is a small price to pay.